pandas/core/base.py

```diff
@@ -733,12 +733,6 @@ def argmax(
         """
         delegate = self._values
```

Comment:
Without using the delegate here and the `nanops.nanargmax` branch below, I'm seeing a 50x perf regression in some ASVs, e.g. `series_methods.NanOps.time_func('argmax', 1000000, 'int8')`:

```
N = 1000000
dtype = 'int8'
s = Series(np.ones(N), dtype=dtype)
```
@jbrockmendel - perhaps that branch should be deeper in the code where the `argmin`/`argmax` implementation is for EAs?

Reply:
Do you mean if you do return `return self.array.argmax(skipna=skipna)` without the EA check?

It looks like sorting.nanargminmax is just a lot slower than the nanops.arg(min|max).  Best case would be to improve sorting.nanargminmax, but failing that I think making NumpyExtensionArray.argmax call the faster version would be viable (need to double-check they are behaviorally equivalent)

Reply:
> Do you mean if you do return `return self.array.argmax(skipna=skipna)` without the EA check?

Correct.

> Best case would be to improve sorting.nanargminmax, but failing that I think making NumpyExtensionArray.argmax call the faster version would be viable

Thanks - will give these a shot.

Reply:
bummer.  thanks for takinga  look